### PR TITLE
Codex attachables

### PR DESCRIPTION
**Pull request by Hughgent**
## About The Pull Request

Automated codex entries for all the attachments. needs some serious review because I'm not certain I got all the nomenclature right.

## Why It's Good For The Game

No more "what does X attachment do?" questions.

## Changelog
:cl: Hughgent
add: Automated codex entries for attachments.
/:cl:

### CODE REVIEW

**Comment by Hughgent** [comment]
image
image
what the numbers and words mean for this, are probably not accurate.

File: code/modules/codex/entries/weapons.dm [hunk 1]
```diff
@@ -1,3 +1,211 @@
...
+	if(flags_attach_features & ATTACH_REMOVABLE)
+		attach_strings += "This can be field stripped off the weapon if needed."
+
+	if(flags_attach_features & ATTACH_ACTIVATION)
+		attach_strings += "This needs to be activated to be used."
+
+	if(flags_attach_features & ATTACH_RELOADABLE)
+		attach_strings += "This can be reloaded with the appropriate ammunition."	
+
+	//if(max_rounds) doesn't work for all the attachements, only the attached_gun ones, UGL, shottie, flamer.
```
Comment (DominikPanic) [inline code/modules/codex/entries/weapons.dm]: You should then attempt to make it work with those, what seems to be the issue exactly?

File: code/modules/codex/entries/weapons.dm [hunk 2]
```diff
@@ -1,3 +1,211 @@
...
+			if(-INFINITY to -30)
+				attach_strings += "Your bullets will fire at the speed of your clicks with this attached."
+			if(-29.99 to -0.01)
+				attach_strings += "You will be able to shoot [delay_mod / 10] faster between clicks with this attached."
+			if(0 to 29.99)
+				attach_strings += "You will be able to shoot [delay_mod / 10] slower between clicks with this attached."
+			if(30 to INFINITY)
+				attach_strings += "You will almost never be able to shoot this weapon. Better pack a book to read inbetween shots."
+
+	//if(burst_delay_mod) no attachment uses this, yet it's defined in gun_attachables
```
Comment (DominikPanic) [inline code/modules/codex/entries/weapons.dm]: Then it's unnecesary commented out code

**Comment by Hughgent** [comment]
image
image

**Comment by Hughgent** [comment]
image

File: code/modules/codex/entries/weapons.dm [hunk 3]
```diff
@@ -1,3 +1,108 @@
...
+	if(damage_falloff_mod)
+		attach_strings += "Damage falloff: [damage_falloff_mod * 100]%"
+	if(burst_scatter_mod)
+		attach_strings += "Scatter chance: [burst_scatter_mod * 100]%"
+	if(silence_mod)
+		attach_strings += "This can silence the weapon if it is attached."
+	if(light_mod)
+		attach_strings += "Light increase: [light_mod] if turned on."
+	if(delay_mod)
+		attach_strings += "delay between shots: [delay_mod / 10]"
```
Comment (DominikPanic) [inline code/modules/codex/entries/weapons.dm]:
```suggestion
		attach_strings += "Delay between shots: [delay_mod / 10]"
```

File: code/modules/codex/entries/weapons.dm [hunk 4]
```diff
@@ -1,3 +1,108 @@
...
+		attach_strings += "Bullet speed: [attach_shell_speed_mod * 100]%"
+	if(movement_acc_penalty_mod)
+		attach_strings += "Running accuracy penalty: [movement_acc_penalty_mod * 100]%"
+	if(max_rounds)
+		attach_strings += "This attachment can hold [max_rounds] of its ammunition."
+	if(scope_zoom_mod)
+		attach_strings += "This has optical glass allowing for magnification and viewing long distances."
+	if(aim_speed_mod)
+		switch(aim_speed_mod)
+			if(SLOWDOWN_ADS_SHOTGUN)
```
Comment (DominikPanic) [inline code/modules/codex/entries/weapons.dm]: Perhaps another sanitization like the INIFITY below?

**Pull request merged**